src/memory/qdrant.rs

```diff
@@ -170,9 +147,9 @@ impl MemorySystem for QdrantMemorySystem {
         let relevant_memories: Vec<RelevantMemory> = search_result
             .iter()
             .map(|point| {
-                let content: String = get_value(&point.payload, "content");
-                let recall: f32 = get_value(&point.payload, "recall");
-                let recency: f32 = get_value(&point.payload, "recency");
+                let json_string = serde_json::to_string(&point.payload).unwrap_or("".to_string());
+
+                let payload: QdrantPayload = serde_json::from_str(&json_string).unwrap();
```

Comment:
I think using the `.unwrap()` is bad, since we're just panicking when we have an error. I would use `?` (i.e. `serde_json::from_str(&json_string)?`.

Reply:
The problem with using `?` is that this is inside a `.map()` function, I cannot propagate the error. Able to come up with a safe, but also much more verbose alternative. Open to other suggestions:

```
let payload: QdrantPayload = match serde_json::from_str(&json_string) {
    Ok(p) => p,
    Err(e) => {
        eprintln!("Failed to parse json_string into QdrantPayload: {}", e);
        return RelevantMemory {
            memory: Memory {
                content: "".to_string(),
                recall: 1.0,
                recency: 1.0,
                embedding: vec![]
            },
            relevance: 0.0,
        }
    }
};
```

Reply:
I think ignoring the error and printing is not the way to go.

Since we're working with Rust iterators, inside of the `map`, you should return a `Result<RelevantMemory, Box<dyn Error>>` inside of the loop. That will give you a `Vec<Result<RelevantMemory, Box<dyn Error>>>`.

Then, you can do `.collect()` to turn that into a `Result<Vec<RelevantMemory>, Box<dyn Error>>`, and then apply the `?` operator on all of that.

Reply:
Good call. Figured out a way to do this. See latest commit.